### ISSUE TITLE
Reimplementing issue

### ISSUE THREAD

**Issue by trungdt880** [SHI-Labs/Smooth-Diffusion#6]
Hi authors, very exciting paper IMO.
I'm currently reimplementing your work but currently facing issue when calculating the Jacobian, since to do that, pytorch need to call the backward function of `attention.backward()` layer, which is not implemented (flash attention and xformers version), I just want to make sure that were you facing the same issue? And your workaround would be using the original (`math`) attention right?
Thanks, look forward to your answer!

**Comment by GiilDe** [comment]
hi @bomcon123456 did you end up implementing this?

**Comment by trungdt880** [comment]
> hi @bomcon123456 did you end up implementing this?

yeah I did, which I largely based on StyleGAN-2-ADA-pytorch repo, but the gradient is so memory costly + due to the lacking of SDPA's double backward function of FlashAttention/xformers (hence using native/math kernel), making the fine-tune very slow. So I wonder how the authors did it.

**Comment by JiayiGuo821** [comment]
Hi @bomcon123456  and @GiilDe, please check `train_smooth_diffusion.py` for our implementation. Thanks!

**Comment by trungdt880** [comment]
awesome <3
